provider/awssd/aws_sd.go

```diff
@@ -409,23 +409,20 @@ func (p *AWSSDProvider) ListServicesByNamespaceID(namespaceID *string) (map[stri
 	return servicesMap, nil
 }
 
-// ListInstancesByServiceID returns list of instances registered in given service.
-func (p *AWSSDProvider) ListInstancesByServiceID(serviceID *string) ([]*sd.InstanceSummary, error) {
-	instances := make([]*sd.InstanceSummary, 0)
+// DiscoverInstancesByServiceID returns list of instances registered in given service.
+func (p *AWSSDProvider) DiscoverInstancesByServiceName(namespaceName, serviceName *string) ([]*sd.HttpInstanceSummary, error) {
```

Comment:
Could we pass in the `context.Context` from `Records()` and pass it through to `DiscoverInstancesWithContext()`?